app/views/books/show.html.erb

```diff
@@ -15,10 +15,10 @@
     Add this book on a bookshelf
   </a>
   <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
-    <% if @bookshelves.empty? %>
+    <% if current_user.bookshelves.empty? %>
```

Comment:
here we missed the if statement for if current_user is nil:
So you can add: (and don't forget the "end" if you want to avoid killing kittens!!) 
```suggestion
    <% if signed_in?  %>
      <% if @bookshelves.empty? %>
```

Reply:
You were right, I added the end so i don't have to kill a kitten.